core/src/test/java/io/grpc/ConnectivityStateInfoTest.java

```diff
@@ -88,6 +88,7 @@ public void equality() {
     assertNotEquals(info4, info6);
 
     assertFalse(info1.equals(null));
-    assertFalse(info1.equals(this));
+    // Extra cast to avoid ErrorProne EqualsIncompatibleType failure
+    assertFalse(((Object) info1).equals(this));
```

Comment:
Not sure.  I think it is expected.equals(actual), but Truth seems to do it 'tother way 'round.  I'm not even sure what this line is trying to prove.  It seems checking with null would do the same thing, and avoid the cast.

Up to you.